sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -927,7 +927,10 @@ export class ServiceBusMessageImpl implements ReceivedMessageWithLock {
     }
     const receiver = this.getReceiverFromContext();
     this.throwIfMessageCannotBeSettled(receiver, DispositionType.complete);
-
+    if (!receiver || !receiver.isOpen()) {
+      await this.backUpMessageSettlementThroughMgmtLink(DispositionStatus.completed);
```

Comment:
A few lines above, we use the $management link to support deferred messages. Can't we use the same instead of using a helper method here?

We should be able to change `if (this._context.requestResponseLockedMessages.has(this.lockToken!)) ` to 
```
if (this._context.requestResponseLockedMessages.has(this.lockToken!) || !receiver || !receiver.isOpen()}} ) 
```

Reply:
Refactored all the settlement methods, please review again :)